web/src/app/services/selected-term.service.ts

```diff
@@ -0,0 +1,106 @@
+                this.terms.set(term.uuid, term);
+                this.termsOrdinal.push(term);
+            });
+            if (localStorage['atFirstTerm'] === 'true') {
+                // if the local storage term was the first term (the most recent)
+                // set the selected term to the most recent instead of restoring
+                // the last setting, which may unnecessarily obfuscate the most
+                // recent term
+                this.setSelectedTermByOrdinal(0);
+            } else {
+                // otherwise restore the selectedTerm from localStorage, verifying its validity
+                // and defaulting to the most recent
+                if (localStorage['selectedTerm'] !== undefined) {
+                    const localTerm = this.terms.get(localStorage['selectedTerm']);
+                    if (localTerm === undefined) {
+                        this.setSelectedTermByOrdinal(0);
+                    } else {
+                        this._currentUUID = localTerm.uuid;
+                        this._currentId = localTerm.id;
+                        this.setSelectedTerm(localTerm.uuid);
+                    }
+                } else {
+                    this.setSelectedTermByOrdinal(0);
+                }
+            }
+        });
+        // internal subscription to term for localstorage
+        this.subscribeToTerm((term: Term) => {
+            this._currentId = term.id;
+            this._currentUUID = term.uuid;
+            localStorage.setItem('selectedTerm', term.uuid);
+            localStorage.setItem('atFirstTerm', `${this._currentOrdinal === 0}`);
+        });
+    }
+
+    /**
+     * Set the selected term by UUID
+     * @param uuid
+     * @return The success of setting the term (returns false if the UUID was invalid)
+     */
+    public setSelectedTerm(uuid: string): boolean {
+        const nTerm = this.terms.get(uuid);
+        if (nTerm !== undefined) {
+            this._currentOrdinal = this.termsOrdinal.findIndex((term) => term.uuid === uuid);
+            this.selectedTerm$.next(nTerm);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Sets the selected term by its ordinal
+     * @param newOrdinal
+     * @return The success of setting the term (returns false if the ordinal was out of bounds)
+     */
+    public setSelectedTermByOrdinal(newOrdinal: number): boolean {
+        if (newOrdinal >= 0 && newOrdinal < this.termsOrdinal.length) {
+            this._currentOrdinal = newOrdinal;
+            this.selectedTerm$.next(this.termsOrdinal[newOrdinal]);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Attaches a subscriber to the selectedTerm, piping errors to console.error.
+     * @param func
+     */
+    public subscribeToTerm(func: (Term) => void): Subscription {
+        return this.selectedTerm$.subscribe(func, (e) => { console.error(e); }, () => {});
```

Comment:
Errors shouldn't really occur, I made a default pipe error to stderr kind of thing here to simplify the subscription call, though alternatively I could change this so that the Subject is exposed and callers can just call `.subscribe` directly.

Reply:
No this is good I like this better than exposing the observable.